tests/unit/test.config.js

```diff
@@ -0,0 +1,162 @@
+
+      const program = makeArgv({
+        userCmd: ['fakecommand', '--source-dir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const argv = program.yargs.exitProcess(false).argv;
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+
+    it('preserves configured value over default', () => {
+      const program = makeArgv({
+        userCmd: ['fakecommand'],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const argv = program.yargs.exitProcess(false).argv;
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const defaultValues = program.defaultValues;
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, configObject.sourceDir);
+    });
+
+    it('preserves a string value on the command line over all others', () => {
+      const cmdLineSrcDir = '/user/specified/source/dir/';
+      const program = makeArgv({
+        userCmd: ['fakecommand', '--sourceDir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const argv = program.yargs.exitProcess(false).argv;
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const defaultValues = program.defaultValues;
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+  });
+
+  describe('loadJSConfigFile', () => {
+    it('throws an error if the config file does not exist', () => {
+      return withTempDir (
+        (tmpDir) => {
+          assert.throws(() => {
+            loadJSConfigFile(tmpDir.path());
+          }, UsageError);
+        });
+    });
+  });
+
+  describe('parseConfig', () => {
+    it('parses the configuration file correctly', () => {
+      const argv = makeArgv({
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      return withTempDir(
+        (tmpDir) => {
+          const configFilePath = path.join(tmpDir.path(), 'config.js');
+          const configFileName = 'config.js';
+          argv.sourceDir = tmpDir.path();
+          fs.writeFileSync(configFilePath,
+          `module.exports = {
```

Comment:
this block should be indented two spaces to show that it's a continuation of the function call